go/database/mpt/visitor.go

```diff
@@ -37,8 +37,9 @@ type NodeVisitor interface {
 }
 
 type NodeInfo struct {
-	Id    NodeId // the ID of the visited node
-	Depth *int   // the nesting level of the visited node, only set for tree visits
+	Id       NodeId // the ID of the visited node
+	Depth    *int   // the nesting level of the visited node, only set for tree visits
+	Embedded bool   // true if this node is embedded in another node, tracked in visitPathTo
```

Comment:
It is not good that this field is only set in `visitPathTo`. 

Visitors should not be aware of the visiting process they are used in. That is part of their design philosophy. 
A fix might be to make it an optional value by turning it into a `*bool` like the `Depth` field above, or `Tribool`, which has the same meaning.

Reply:
I would consider it ok when the value is not set and then the default is false. But ok, I changed it to Tribool